From b5745592fba1d0d8e35c348808a272b3de734e13 Mon Sep 17 00:00:00 2001
From: Ryan QIAN <b32804@freescale.com>
Date: Wed, 16 Jan 2013 10:23:28 +0800
Subject: [PATCH 0751/1543] ENGR00240226-07 mmc: add delay after CMD6 befoer
 sending CMD13 for sandisk

commit 2d986e1699ae3c4cb65374cc07ddf6cd4db5c1cf from
git://git.freescale.com/imx/linux-2.6-imx.git

- Some sandisk emmc cards need certain delay befor sending cmd13 after
cmd6. Original CR: ENGR174296 (commit: fd031f9)

Acked-by: Aisheng Dong <b29396@freescale.com>
Signed-off-by: Ryan QIAN <b32804@freescale.com>
(cherry picked from commit f942bf1db36355d46f38792601594949f3f2c71b)
---
 drivers/mmc/core/mmc_ops.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/core/mmc_ops.c b/drivers/mmc/core/mmc_ops.c
index e5b5eeb..b2680c9 100644
--- a/drivers/mmc/core/mmc_ops.c
+++ b/drivers/mmc/core/mmc_ops.c
@@ -445,6 +445,12 @@ int __mmc_switch(struct mmc_card *card, u8 set, u8 index, u8 value,
 		return 0;
 
 	/*
+	 * WORKAROUND: for Sandisk eMMC cards, it might need certain delay
+	 * before sending CMD13 after CMD6
+	 */
+	mdelay(1);
+
+	/*
 	 * Must check status to be sure of no errors
 	 * If CMD13 is to check the busy completion of the timing change,
 	 * disable the check of CRC error.
-- 
1.7.5.4

